src/dataflow-types/src/types/sources.proto

```diff
@@ -62,6 +63,22 @@ message ProtoDebeziumTransactionMetadata {
     uint64 data_transaction_id_idx = 8;
 }
 
+message ProtoDebeziumMode {
+    message ProtoFullInRange {
+        ProtoDebeziumDedupProjection projection = 1;
+        mz_repr.chrono.ProtoNaiveDateTime pad_start = 2;
```

Comment:
shouldn't this be `optional` to make it more clear its different?

Reply:
All types are always optional by default in proto3, and map to an `Option<T>` in the Rust side by `prost`!

Technically all fields are always optional, but for scalar types the `optional` annotation actually changes how they are represented Rust-side (otherwise they would just deserialize to the `Default` value of the type, following Go semantics), more details [here](https://github.com/tokio-rs/prost#field-modifiers).

Reply:
It sounds like for a struct type like this, `optional` doesn't change anything, but I do think it still make this case more clear!